x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -3514,6 +3514,49 @@ public void testBucketWithAggExpression() {
         assertThat(agg.groupings().get(0), is(ref));
     }
 
+    public void testBucketWithNonFoldingArgs() {
```

Comment:
nit: wouldn't these be better placed in the `VerifierTests`? I wouldn't know to look here for these tests (except when they're failing :) )

Reply:
It turns out we do the validation past optimisations, to be able to take the folding into account. (Not sure if we'll ever need this for anything else than supporting ROW, but that's how it now works.)